### ISSUE TITLE
BaseHtml activeTextarea bug

### ISSUE THREAD

**Issue by hiscaler** [yiisoft/yii2#8585]
~~~php
$value = static::getAttributeValue($model, $attribute);
~~~
will generate
~~~txt
<textarea id="interviewform-questionanswer" class="form-control" name="InterviewForm[questionAnswer][]" value="a"></textarea>
~~~
maybe change to 
~~~php
$value = isset($options['value']) ? $options['value'] : static::getAttributeValue($model, $attribute);
~~~
will generate correct code
~~~txt
<textarea id="interviewform-questionanswer" class="form-control" name="InterviewForm[questionAnswer][]" value="a">a</textarea>
~~~

**Comment by klimov-paul** [comment]
`Html::activeTextArea()` works correctly. There is a unit test, which proves this:
https://github.com/yiisoft/yii2/blob/master/tests/framework/helpers/HtmlTest.php#L849

**Comment by hiscaler** [comment]
No, if the textarea name is array, generate html is wrong.

**Comment by cebe** [comment]
can you show the code you use to generate the textarea?

**Comment by hiscaler** [comment]
~~~php
<div id="panel-customize" class="panel" style="<?= $tab == 'customize' ? '' : 'display: none' ?>">
                <?php
                $questions = InterviewQuestion::getItems();
                $questionAnswers = $model->questionAnswers;
                foreach (InterviewQuestion::typeOptions() as $groupId => $groupName):
                    ?>
                    <div class="wrapper">
                        <fieldset>
                            <legend class="group-name"><?= $groupName ?></legend>
                            <?php
                            if (isset($questions[$groupId])) {
                                foreach ($questions[$groupId] as $id => $value):
                                    ?>
                                    <fieldset>
                                        <legend><?= $value ?></legend>
                                        <?php
                                        echo $form->field($model, 'questionId[]')->hiddenInput(['value' => $id])->label(false);
                                        echo $form->field($model, 'questionAnswer[]', ['template' => "<dl><dt>{label}</dt>\n<dd>{input}\n{hint}\n{error}</dd>"])
                                                ->textarea(['value' => isset($questionAnswers[$id]) ? $questionAnswers[$id] : null])
                                                ->label(false);
                                        ?>
                                    </fieldset>
                                    <?php
                                endforeach;
                            } else {
                                echo Html::tag('div', 'Nothing', ['class' => 'notice']);
                            }
                            ?>
                        </fieldset>
                    </div>
                <?php endforeach; ?>
            </div>
~~~

**Comment by hiscaler** [comment]
I must manual setting value. Because field is dynamic.

**Comment by klimov-paul** [comment]
Issue resolved by commit 0874ffa
